Src/PerformanceCollector/Shared/Implementation/QuickPulse/QuickPulseEventSource.cs

```diff
@@ -0,0 +1,135 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.Implementation.QuickPulse
+{
+    using System;
+#if !NET40
+    using System.Diagnostics.Tracing;
+#endif
+
+#if NET40
+    using Microsoft.Diagnostics.Tracing;
+#endif
+
+    [EventSource(Name = "Microsoft-ApplicationInsights-Extensibility-PerformanceCollector-QuickPulse")]
+    internal sealed class QuickPulseEventSource : EventSource
+    {
+        private static readonly QuickPulseEventSource Logger = new QuickPulseEventSource();
+
+        private QuickPulseEventSource()
+        {
+            this.ApplicationName = this.GetApplicationName();
+        }
+
+        public static QuickPulseEventSource Log
+        {
+            get
+            {
+                return Logger;
+            }
+        }
+
+        public string ApplicationName { [NonEvent]get; [NonEvent]private set; }
+
+        #region Infra init - success
+
+        [Event(1, Level = EventLevel.Informational, Message = @"QuickPulse infrastructure is being initialized. {0}")]
+        public void ModuleIsBeingInitializedEvent(
+            string message,
+            string dummy = "dummy",
+            string applicationName = "dummy")
+        {
+            this.WriteEvent(1, message, this.ApplicationName);
+        }
+
+        [Event(3, Level = EventLevel.Informational, Message = @"Performance counter {0} has been successfully registered with QuickPulse performance collector.")]
+        public void CounterRegisteredEvent(string counter, string applicationName = "dummy")
+        {
+            this.WriteEvent(3, counter, this.ApplicationName);
+        }
+        #endregion
+
+        #region Infra init - failure
+        [Event(5, Keywords = Keywords.UserActionable, Level = EventLevel.Warning, Message = @"Performance counter {1} has failed to register with QuickPulse performance collector. This might happen whenever an application is running on a platform that doesn't provide access to performance counters. Technical details: {0}")]
```

Comment:
Warning will not be sent to a customer. What's the point making it user actionable?